src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let joinedArray = '';
+
+    if (this[0] === null || this[0] === undefined) {
+      joinedArray = '';
+    } else {
+      joinedArray += this[0];
+    }
```

Comment:
You can also write this code:
```suggestion
    if (this[0] !== null && this[0] !== undefined) {
      joinedArray += this[0];
    }
```

Reply:
Let's get rid of this `if` at all